srtcore/core.cpp

```diff
@@ -8652,13 +8652,9 @@ void CUDT::processCtrl(const CPacket &ctrlpkt)
         m_bBroken        = true;
         m_iBrokenCounter = 60;
 
-        // Signal the sender and recver if they are waiting for data.
-        releaseSynch();
-        // Unblock any call so they learn the connection_broken error
-        s_UDTUnited.m_EPoll.update_events(m_SocketID, m_sPollID, SRT_EPOLL_ERR, true);
-
-        CGlobEvent::triggerEvent();
-
+        // This does the same as it would happen on connection timeout,
+        // just we know about this state prematurely thanks to this message.
+        updateBrokenConnection(SRT_ECONNLOST);
```

Comment:
Connection hook will now be also called at shutdown `UMSG_SHUTDOWN`? Is this correct?

Reply:
Correct. The SHUTDOWN situation differs to TIMEOUT only by the fact that SHUTDOWN succeeded to be received by the agent party and it should be handled the same way.

Reply:
From [API-functions.md](https://github.com/Haivision/srt/blob/master/docs/API-functions.md#srt_connect_callback) about `srt_connect_callback`:
> This call installs a callback hook, which will be executed on a given clr socket just after the pending connection situation in the background has been resolved (that is, when the connection succeeded or failed). 

SHUTDOWN may arrive:
- right after HS response, when callback makes sense,
- in the end of transmission if the peer closes connection, then callback does not look reasonable.

Reply:
Ok, I understand what you mean. The real intention for the connect callback is to catch the moment when the socket got broken and is about to be removed in order to collect the information before the socket disappears and the information about the broken connection is lost. This is important regardless as to whether it didn't ever connect due to rejection, or got broken after being functional for some time.

In the light of this, it's actually the description not exactly correct.

Reply:
Still topical.
Connection callback is now called in SHUTDOWN and not called anymore on a successfull connection.
Fix API-Functions.md